client/components/TestRun/index.jsx

```diff
@@ -656,10 +659,14 @@ const TestRun = () => {
                 id,
                 output: output,
                 unexpectedBehaviors: unexpectedBehaviors,
-                assertionResults: assertionResults.map(({ id, passed }) => ({
-                    id,
-                    passed
-                }))
+                assertionResults: assertionResults
+                    // All assertions are always being passed from the TestRenderer results, but in the instance where
+                    // there is a 0-priority exception, and id won't be provided and that cannot be saved
```

Comment:
Is this taking advantage of a side effect or is this the strategy used for communicating that an assertion is excluded? It seems easy to misunderstand or forget. I would expect them to be passed prefiltered but I might be missing something.

Reply:
It's been some time now, so I'll have to look back into why exactly it was done that way. But from what I can remember, the `TestRenderer` still technically requires the index of that excluded assertion to properly determine which assertions are which, but at some point. It definitely had to be removed before saving and those excluded ones had no ids. That was a part of fulfilling [this request](https://github.com/w3c/aria-at-app/pull/863#pullrequestreview-1778516739) with https://github.com/w3c/aria-at-app/pull/863/commits/5c4ec6a5568b685e1c7f044f7556912c32b811a6.

> Is this taking advantage of a side effect or is this the strategy used for communicating that an assertion is excluded?

So seems to be a bit of both.

This point in the code looks like the safest place to filter them out, as the single point before being passed to the server/db. The comment could be much clearer as well.

Reply:
Oh okay. The background is helpful. I'll leave it to you as to whether you want to change the comment or address it any other way.